prov/mrail/src/mrail_ep.c

```diff
@@ -261,6 +261,18 @@ static ssize_t mrail_recv(struct fid_ep *ep_fid, void *buf, size_t len,
 				 mrail_ep->util_ep.rx_op_flags, FI_MSG);
 }
 
+static ssize_t mrail_recvmsg(struct fid_ep *ep_fid, const struct fi_msg *msg,
+		uint64_t flags)
+{
+	struct mrail_ep *mrail_ep = container_of(ep_fid, struct mrail_ep,
+					     util_ep.ep_fid.fid);
+
+	return mrail_recv_common(mrail_ep, &mrail_ep->recv_queue,
+				 (struct iovec *)msg->msg_iov, msg->iov_count,
+				 msg->context, msg->addr, 0, 0,
+				 mrail_ep->util_ep.rx_op_flags | flags, FI_MSG);
```

Comment:
Shouldn't the flags be always a combination of the default + any flags specific to the current operation? The only exception is FI_COMPLETION which would depend on FI_SELECTIVE_COMPLETION but that's handled separately.

Reply:
The default flags should apply to fi_recv and fi_recvv, which don't take flags as input.  Since fi_recvmsg takes the flags argument, it replaces the default flags.  Otherwise, you could never get rid of a flag that you don't want for a specific call.